### PR TITLE
Add ability to start and terminate tested app (closes #1047). Move to yarn for CI (closes #864)

### LINKED ISSUE #864: Use yarnpkg on CI
https://yarnpkg.com/
Currently blocked by https://github.com/yarnpkg/yarn/issues/684

### LINKED ISSUE #1047: `--app-command` option
### Are you requesting a feature or reporting a bug?

feature

### What is the current behavior?

There is no way to run tested server app and terminate it once tests finish.

### What is the expected behavior?

Provide `--app-command` option which will allow to start test server and automatically terminate it, once testing is finished, e.g.:
```sh
testcafe chrome --app-command "node server.js" tests/
```

### PR DESCRIPTION
**Pull request by inikulin**
## Feature summary

This PR adds `-a, --app-command` and `--app-init-delay` CLI options as well as `Runner.startApp(command, initDelay)` method for the programmatic API. TestCafe will start the app specified by command (it can be any shell command) and automatically terminate it once the testing finished. Additionally you can specify app init delay - the amount the time that TestCafe will wait after command execution before starting the testing. Default value is `1000` ms. Note that TestCafe adds `node_modules/.bin` directory to the `PATH` for the app command, so like with [npm scripts](https://docs.npmjs.com/cli/run-script) you can use locally-installed binaries.

\cc @AlexanderMoskovkin @churkin @VasilyStrelyaev

### CODE REVIEW

**Comment by inikulin** [comment]
BTW, we need to change all recipes to use this approach instead of concurrently or shell scripts \cc @DevExpress/testcafe-docs

File: src/cli/argument-parser.js [hunk 1]
```diff
@@ -98,6 +98,8 @@ export default class CLIArgumentParser {
             .option('-T, --test-grep <pattern>', 'run only tests matching the specified pattern')
             .option('-f, --fixture <name>', 'run only fixtures with the specified name')
             .option('-F, --fixture-grep <pattern>', 'run only fixtures matching the specified pattern')
+            .option('-a, --app <command>', 'specify tested app startup command')
+            .option('--app-init-delay <ms>', 'specify time required for the initialization of the tested app before testing starts')
```
Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]:
```
            .option('-a, --app <command>', 'launch the tested app using the specified command before running tests')
            .option('--app-init-delay <ms>', 'specify how much time it takes for the tested app to initialize')
```

File: src/errors/runtime/message.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ export default {
     portNumberIsNotInteger:                         'A port number should be a valid integer.',
     selectorTimeoutIsNotAnInteger:                  'Selector timeout should be an integer.',
     assertionTimeoutIsNotAnInteger:                 'Assertion timeout should be an integer.',
+    appInitDelayIsNotAnInteger:                     'Tested app initialization delay should be an integer.',
+    testedAppFailedWithError:                       'Tested app failed with the error:\n\n{errMessage}',
```
Comment (VasilyStrelyaev) [inline src/errors/runtime/message.js]:
```
    testedAppFailedWithError:                       'Tested app failed with an error:\n\n{errMessage}',
```

**Comment by inikulin** [comment]
FPR

**Pull request merged**